### ISSUE TITLE
tf.random.uniform array minval/maxval with dtype=tf.int*

### ISSUE THREAD

**Issue by chrism0dwk** [tensorflow/tensorflow#39814]
Hi,

The current implementation of `tf.random.uniform` allows for non-scalar `minval` and `maxval` if `dtype=tf.float32` (or `tf.float64`).  However, this is not the case for `dtype=tf.int32` (or `tf.int64`).

e.g.
```python
tf.random.uniform(shape=(), minval=[0, 0], maxval=[10, 10], dtype=tf.int32)
ValueError: minval must be a scalar; got a tensor of shape [2] ....
```
generates a `ValueError` whereas
```python
tf.random.uniform(shape=(), minval=[0., 0.], maxval=[10., 10.], dtype=tf.float32)
tf.Tensor([5.918947 5.918947], shape=(2,), dtype=float32)
```
works as expected.

The documentation doesn't indicate that the integer version of `tf.random.uniform` has this restriction.  Is it expected behaviour, and if so what is the workaround?

Thanks,

Chris

----
TF version 2.3.0-dev20200521 (tf-nightly)
Python 3.7.6 (Anaconda 3)

**Comment by yongtang** [comment]
@chrism0dwk The ValueError should be the correct behavior, as `shape=()` expect minval and maxval to be a scalar.

The behavior of the float32 type is actually incorrect, and should have thrown out the same error as well. The issue is being fixed in #34363 #38585

**Comment by chrism0dwk** [comment]
Ah, I see.  Thanks.  From the doc of `minval` and `maxval`, "for integer types, broadcasting is not supported, so it needs to be a scalar".  I guess this means that `minval` and `maxval` are therefore intended to be scalars at least for Uniform integers.
